.github/workflows/test.yml

```diff
@@ -98,4 +98,37 @@ jobs:
       - name: Test
         run: |
           python -m pip install -r tests/python/requirements.txt
-          PYTHONPATH=$(pwd)/build/dist/hybridpython python -m pytest -v
\ No newline at end of file
+          PYTHONPATH=$(pwd)/build/dist/hybridpython python -m pytest -v
+
+  disallowed_pip_install:
+    name: Disallowed pip install
+    runs-on: ubuntu-latest
+    container:
+      image: aswf/ci-vfxall:2022-clang14.3
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v4
+
+      - name: Install Traitgen
+        run: python -m pip install openassetio-traitgen==1.0.0a6
+
+      - name: Set Python Root Dir
+        run: echo "Python_ROOT_DIR=$(python -c 'import sys; print(sys.prefix)')" >> $GITHUB_ENV
```

Comment:
It's neccesary because we're in the docker container, so it dosen't get set.

Reply:
Ah, in which case, does `setup-python` actually do anything? Since it's our Docker image, do we even need to programmatically determine where Python is installed?  i.e. we could hardcode it (presumably it's `/usr` or less likely `/usr/local`).

Reply:
Does seem odd that the container is set for the whole workflow, and `setup-python` is a step within that workflow, yet doesn't set the env var?  Must be conditional logic in the action?

Reply:
setup-python _may_ not do anything, I didn't give it a lot of thought once it was functioning. Will see.

Reply:
I wonder if setup-python just aborts early without doing anything if it detects the environment is already setup?

Reply:
Yep, we're all good without setup python as you thought. https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/66/commits/afe0d9a0a0e3f49f9fd4a2e8f985007e509f3404

I'm not gonna stop programatically getting the python location, I think it's better done that way.